### PR TITLE
PXB-2662 using pagetracking pxb crashes during prepare, cannot apply …

### PR DESCRIPTION
**Pull request by rahulmalik87**
…an insert redo on page
https://jira.percona.com/browse/PXB-2662
follow up fix. Declaration of extern variable in header file.

### CODE REVIEW

File: storage/innobase/xtrabackup/src/read_filt.cc [hunk 1]
```diff
@@ -249,8 +247,9 @@ static void rf_page_tracking_get_next_batch(xb_fil_cur_t *cursor,
    * backup start_lsn instead of last backup end_lsn.
    * We read all pages by setting read_batch_len to the size of file */
 
-  /* if DDL happened on the table after the checkpoint LSN we do full scan on
-  that table. index_load_map is poped during the first scan of redo */
+  /* if inplace DDL having MLOG_ADD_INDEX happened on the table after the
```
Comment (satya-bodapati) [inline storage/innobase/xtrabackup/src/read_filt.cc]: If INPLACE DDLs that generate MLOG_INDEX_LOAD happen on table after the checkpoint.."

File: storage/innobase/xtrabackup/src/read_filt.cc [hunk 2]
```diff
@@ -249,8 +247,9 @@ static void rf_page_tracking_get_next_batch(xb_fil_cur_t *cursor,
...
+  checkpoint LSN we do full scan on that table. index_load_map is poped during
```
Comment (satya-bodapati) [inline storage/innobase/xtrabackup/src/read_filt.cc]: poped? u mean populated?

**Review by satya-bodapati — CHANGES_REQUESTED** [review]

File: storage/innobase/include/log0recv.h [hunk 3]
```diff
@@ -52,6 +52,14 @@ class PersistentTableMetadata;
...
 struct recv_addr_t;
 
+#ifdef XTRABACKUP
+/** map of tablespace_id, that experienced an inplace DDL during a backup op */
+extern std::map<space_id_t, bool> index_load_map;
+extern bool opt_page_tracking;
+extern char *xtrabackup_incremental;
+extern lsn_t incremental_start_checkpoint_lsn;
+#endif /* XTRABACKUP */
+
```
Comment (altmannmarcelo) [inline storage/innobase/include/log0recv.h]: those all comes from `xtrabackup.cc` their logical place is where they are declared and not where they are used.
Xtrabackup uses xtrabackup.h as header file for xtrabackup related source code( the ones under storage/innobase/xtrabackup/src`) and xb0xb.h for shared code between xtrabackup and server/innodb. 
Those extern's belong to xb0xb.h

File: storage/innobase/include/xb0xb.h [hunk 4]
```diff
@@ -26,6 +26,7 @@ extern bool innodb_checksum_algorithm_specified;
 
 extern bool opt_lock_ddl_per_table;
 extern bool redo_catchup_completed;
+extern lsn_t incremental_start_checkpoint_lsn;
```
Comment (altmannmarcelo) [inline storage/innobase/include/xb0xb.h:31]: you already externing it on log0recv.h

**Review by altmannmarcelo — CHANGES_REQUESTED** [review]

File: storage/innobase/xtrabackup/src/read_filt.cc (hunk 1 above)
Comment (rahulmalik87) [reply storage/innobase/xtrabackup/src/read_filt.cc]: fixed

File: storage/innobase/xtrabackup/src/read_filt.cc (hunk 2 above)
Comment (rahulmalik87) [reply storage/innobase/xtrabackup/src/read_filt.cc]: fixed

File: storage/innobase/include/log0recv.h (hunk 3 above)
Comment (rahulmalik87) [reply storage/innobase/include/log0recv.h]: fixed

File: storage/innobase/include/xb0xb.h (hunk 4 above)
Comment (rahulmalik87) [reply storage/innobase/include/xb0xb.h:31]: fixed

File: storage/innobase/xtrabackup/src/read_filt.cc [hunk 5]
```diff
@@ -249,8 +247,9 @@ static void rf_page_tracking_get_next_batch(xb_fil_cur_t *cursor,
    * backup start_lsn instead of last backup end_lsn.
    * We read all pages by setting read_batch_len to the size of file */
 
-  /* if DDL happened on the table after the checkpoint LSN we do full scan on
-  that table. index_load_map is poped during the first scan of redo */
+  /* if inplace DDLs that generated MLOG_ADD_INDEX happened on the table after
```
Comment (satya-bodapati) [inline storage/innobase/xtrabackup/src/read_filt.cc]:
```suggestion
  /* if inplace DDLs that generated MLOG_INDEX_LOAD happened on the table after
```

**Review by satya-bodapati — CHANGES_REQUESTED** [review]

File: storage/innobase/xtrabackup/src/read_filt.cc (hunk 5 above)
Comment (rahulmalik87) [reply storage/innobase/xtrabackup/src/read_filt.cc]: fixed

**Review by satya-bodapati — APPROVED** [review]

**Review by altmannmarcelo — APPROVED** [review]
LGTM

**Pull request merged**